### ISSUE TITLE
Footnotes not WCAG 2.1 compliant

### ISSUE THREAD

**Issue by JamesCatt** [wet-boew/wet-boew#8528]
FYI the [footnotes pattern](https://wet-boew.github.io/v4.0-ci/demos/footnotes/footnotes-en.html) is not compliant with WCAG 2.1, specifically [SC 2.5.3 Label in Name](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html), since the hidden text ("footnote") is at the beginning of the link text. The simplest solution to this would be to change the hidden text to follow the number. For example:

```
<sup id="fn1-rf"><a class="fn-lnk" href="#fn1">1<span class="wb-inv"> (footnote)</span></a></sup>
```

**Comment by Christopher-O** [comment]
Would this also be an issue with buttons that only have an icon displayed (like a garbage can, or a printer)? 
http://wet-boew.github.io/wet-boew-styleguide/v4/design/buttons-en.html#btn-types

**Comment by JamesCatt** [comment]
@duboisp Sorry for the delay on this. PR's almost ready. A couple of notes:

- For the French footnotes, I went with the pattern `<a class="fn-lnk" href="#fn2">1<span class="wb-inv">, note de bas de page</span></a>`
- For the reference return links, I went with the pattern `<a href="#fn1-rf">1<span class="wb-inv">, return to footnote referrer</span></a>` in English, and `<a href="#fn2-1-rf">1<span class="wb-inv">, retour à la <span>première</span> référence de la note de bas de page</span></a>` for French.
- I added the explanatory note as you asked, in a section titled *Changelog*. My French isn't good enough to translate this, so I added it in a *Needs translation* section as has been done elsewhere
- I've only modified the Handlebars files, as I believe this change shouldn't require any modification to the javascript.

Let me know if I missed anything.

> Would this also be an issue with buttons that only have an icon displayed (like a garbage can, or a printer)?
> http://wet-boew.github.io/wet-boew-styleguide/v4/design/buttons-en.html#btn-types

@Christopher-O I don't believe so. SC 2.5.3 is just about ensuring that visible text (if any) matches the accessible name. If there's no visual text then it's not relevant.

**Comment by EricDunsworth** [comment]
IMO the status quo could arguably pass 2.5.3.

**Quote of [SC 2.5.3's description](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html):**
> Success Criterion 2.5.3 Label in Name (Level A): For user interface components with labels that include text or images of text, the name contains the text that is presented visually.
> 
> > A best practice is to have the text of the label at the start of the name.

If I were to strictly read that on its own, my take on it would be that the requirement is for the text that's being presented visually to exist somewhere within the name. That's already being done in the pre-existing footnote links (even though the visible text isn't situated at the beginning of the link text).

The links obviously don't meet the best practice. But given that it was presented as a best practice (as opposed to a requirement), I don't think the status quo would clearly represent a failure.

On another note, SC 2.5.3 is seriously lacking in techniques at this time:
* Its sufficient techniques section currently only contains a placeholder for a technique called "Ensuring that visible labels match their accessible names", which sounds much more strict than what's described in SC 2.5.3's description.
* The advisory technique and failures are currently also mostly placeholders.
* The only documented failure technique is called "Failure due to "accessible name" not containing the visible label text". The pre-existing footnote links don't fail its test procedure, nor do they match the situations covered in any of its examples.

Maybe there'd be value in revisiting this once SC 2.5.3's techniques are more fully-developed?

**Comment by JamesCatt** [comment]
> If I were to strictly read that on its own, my take on it would be that the requirement is for the text that's being presented visually to exist somewhere within the name. That's already being done in the pre-existing footnote links (even though the visible text isn't situated at the beginning of the link text).
> 
> The links obviously don't meet the best practice. But given that it was presented as a best practice (as opposed to a requirement), I don't think the status quo would clearly represent a failure.

@EricDunsworth Correct--I realized this belatedly. I also just tested with Dragon Naturally Speaking and the existing pattern works when speaking "click footnote".

With that in mind, feel free to close the issue and/or PR as you see fit.

**Comment by Christopher-O** [comment]
@JamesCatt  Thank you for the response.  I misread "images as text" as images instead of text.  Though the issue may not violate this particular WCAG issue directly, I still think it could be an accessibility issue with these types of readers.  For example, if a webmaster tagged a button with a garbage can with the text "remove"...many people think of a garbage can as "delete" and if the person is saying "delete" nothing will happen.  They would have to know that the button link is tagged with the label "remove".
